### PR TITLE
Documentation: Move "Modules" into "Configuration"

### PR DESCRIPTION
**Pull request by Arnei**
The differentiation between "modules" and "configuration" in the documentation may been useful in the past, but is not anymore today. From an admin perspective, I don't care if the thing I'm configuring is a "module" or not, and either end up looking through both "modules" and "configuration" or just using the search function.

This PR gets rid of the "modules" tab by moving all its pages to the "configuration" tab. It also takes the opportunity to consolidate information and remove/rework outdated information

### CODE REVIEW

**Comment by Arnei** [comment]
Don't understand why tests are failing, they run fine locally.

**Comment by KatrinIhler** [comment]
> Don't understand why tests are failing, they run fine locally.

Could it be that you're running a different version of markdownlint locally that is missing certain checks? I ran into this problem recently with ansible linting.

**Comment by Arnei** [comment]
I did run `npm install` before `npm start`, so there shouldn't be any major version differences.

**Comment by KatrinIhler** [comment]
Hm, then I don't know. I guess you could always try to fix the stuff the pipeline is complaining about :man_shrugging:

**Comment by ziegenberg** [comment]
Looking at `docs/checkstyle/check-docs.sh`, it fails while running `mkdocs build` for the developer docs.

```
Building developer documentation…
INFO    -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
WARNING -  A relative path to 'architecture/new-admin-ui.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A relative path to 'architecture/stream-security-config.md' is included in the 'nav' configuration, which is not found in the documentation files.
developer did not build correctly!
```

File: docs/guides/developer/mkdocs.yml
Comment (ziegenberg) [inline docs/guides/developer/mkdocs.yml:1]: The two changes in this file let `mkdocs build` trip over its on feet.

**Review by ziegenberg — CHANGES_REQUESTED** [review]

File: docs/guides/developer/mkdocs.yml
Comment (Arnei) [reply docs/guides/developer/mkdocs.yml:1]: Tfw you think you didn't touch the developer docs so you come to the conclusion that those warnings couldn't possibly be related *facepalm*. Thanks for pointing this out, did the trick.

Comment (ziegenberg) [reply docs/guides/developer/mkdocs.yml:1]: Looks good now :)

File: docs/guides/admin/docs/index.md [hunk 1]
```diff
@@ -28,19 +28,17 @@ The Opencast Release Documentation is the official Opencast documentation for ea
...
+    * [Media Module](configuration/mediamodule.configuration.md)
+    * [Metrics (OpenMetrics, Prometheus)](configuration/metrics.md)
+    * [New Admin UI](configuration/admin-ui/new-admin-ui.md)
+    * [Stream Security](configuration/stream-security/stream-security-overview.md)
+    * [Stand-alone Video Editor](configuration/videoeditor.frontend.md)
+    * [Studio](configuration/studio.md)
+    * [Text Extraction](configuration/textextraction.md)
+    * [Video Segmentation](configuration/videosegmentation.md)
+    * [YouTube Publication](configuration/youtubepublication.md)
     * [more...](configuration/index.md)
```
Comment (gregorydlogan) [inline docs/guides/admin/docs/index.md]: I don't dislike this as much as I thought I would, but I would definitely edit the list of default config items shown to shorten it up.

What about this?

```suggestion
    * [Plugin Management](configuration/plugin-management.md)
    * [Atom and RSS Feed](configuration/atomrss.md)
    * [Metrics (OpenMetrics, Prometheus)](configuration/metrics.md)
    * [New Admin UI](configuration/admin-ui/new-admin-ui.md)
    * [Stream Security](configuration/stream-security/stream-security-overview.md)
    * [Stand-alone Video Editor](configuration/videoeditor.frontend.md)
    * [Studio](configuration/studio.md)
    * [more...](configuration/index.md)
```

(drops s3 dist, media module, text extraction, video segmentation, and youtube publication).  I could see arguments for removing other things (admin ui, feeds, metrics) too, but this at least shortens the list down to keep the initial shock of the list size down.

Comment (Arnei) [reply docs/guides/admin/docs/index.md]: I have no opinion on what is and is not shown in that list (it always seemed somewhat superfluous to me), so I'll go along with your suggestion if no one disagrees.

**Comment by Arnei** [comment]
@gregorydlogan, think this could get approved today? I'd like to have this in OC15 (and not have to resolve conflicts everytime someone pushes a change to the docs).

**Review by ziegenberg — APPROVED** [review]

**Comment by gregorydlogan** [comment]
LGTM, merging.

**Pull request merged**